[swx0] iP

swx0 opened the pull request:
## Duke

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

swx0 committed:
Add greet, echo, exit

swx0 committed:
added Task, TaskList, statuses for message responses

swx0 committed:
Add marking task as done or undone

swx0 committed:
Add ToDos, Events, Deadlines tasks

swx0 committed:
Added header and end message lines for task messages

swx0 committed:
Added testing of text UI

swx0 committed:
Added InvalidInputException for unknown user input

swx0 committed:
Added InvalidInputException for unknown user input

swx0 committed:
Added InvalidDeadlineException, InvalidEventException and InvalidTodoException to handle errors for task inputs

swx0 committed:
Fixed formatting of Event toString method

swx0 committed:
Updated TestUITesting for testing handling of errors

swx0 committed:
Added delete tasks

swx0 committed:
Refactor generateMessage functions

swx0 committed:
Add saving and loading of tasks in disk

swx0 committed:
Change stored date and time from String to LocalDateTime objects

swx0 committed:
Fix merge conflict

swx0 committed:
Refactor MessageProcessor to MessageGenerator and Parser objects

swx0 committed:
Add packages

swx0 committed:
removed text file from data folder

swx0 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

swx0 committed:
Merge add-gradle-support branch

swx0 committed:
Add duke package

swx0 committed:
Add JUnit tests

swx0 committed:
Delete extra imports from JUnit test

swx0 committed:
Reformat code

swx0 committed:
Add JavaDoc for classes and methods

swx0 committed:
Add finding tasks by keyword

swx0 committed:
Fix merge conflicts

swx0 committed:
Fix merge conflicts

swx0 committed:
Fix switch indentations

swx0 committed:
Add GUI

0x787af25e commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,79 @@
+        }
+        String action = messageSplit[0];
+
+        if ((action.equals("mark") || action.equals("unmark"))) {
+            return true;
+        }
+        return false;
```
I would have
```return (action.equals("mark") || action.equals("unmark"))```

0x787af25e commented on src/main/java/duke/message/MessageGenerator.java:
```diff
@@ -0,0 +1,145 @@
+     * @return String consisting of the header, associated task and end.
+     */
+    String generateTaskMessage(MessageStatus status, Task task) {
+        String heading = "";
+        String end = "";
```
Should be in all caps, and marked with the final specifier.

0x787af25e commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,31 @@
+ */
+public class Task {
+    private String content;
+    boolean done = false;
```
Should be `isDone`.

0x787af25e reviewed (commented):
Overall a very good effort with code consistent with the conventions.

0x787af25e commented on data/tasklist.txt:
```diff
@@ -0,0 +1,7 @@
+mark todo sdsadsd
```
Don't commit stuff that should be placed in `.gitignore`

ChickenChiang commented on src/main/java/duke/GUI/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+package duke.gui;
+
+import java.io.IOException;
```
Good use of specific imports instead of using wildcards

ChickenChiang commented on src/main/java/duke/GUI/MainWindow.java:
```diff
@@ -0,0 +1,55 @@
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+    }
+
+    public void setDuke(Duke d) {
```
Perhaps a descriptive header is needed here since it is a public method?

ChickenChiang commented on src/main/java/duke/GUI/Main.java:
```diff
@@ -0,0 +1,32 @@
+package duke.gui;
+
+import java.io.IOException;
+
```
Good ordering of import statements, but perhaps line 4 can be removed to keep the formatting consistent?

swx0 committed:
change mainClassName in build.gradle for JavaFX

swx0 committed:
Add assertions
To check if assumptions hold
Ensure the correctness of the program. If assertion fails, program will terminate

swx0 committed:
Refactor code
To improve readability and simplify logic

swx0 committed:
Merge pull request #1 from swx0/branch-A-Assertions

Add assertions

swx0 committed:
Merge branch 'master' into branch-A-CodeQuality

swx0 committed:
Merge pull request #2 from swx0/branch-A-CodeQuality

Refactor code

swx0 committed:
Setup CI

swx0 committed:
Add sorting tasks by start date

swx0 committed:
Update gitignore

swx0 committed:
Delete tasklist.txt

swx0 committed:
Update readme

swx0 committed:
Update readme
